File: frontend/src/types/action.ts
export type PlatformType =
  | 'Power Automate Cloud'
  | 'Power Automate Desktop'
  | 'Hybrid'
  | 'Manual Review';

export type ComplexityType = 'Low' | 'Medium' | 'High';

export type MigrationStrategyType =
  | 'Direct Mapping'
  | 'Equivalent Connector'
  | 'Cloud Replacement'
  | 'Desktop Replacement'
  | 'Hybrid Implementation'
  | 'Custom API'
  | 'Custom Script'
  | 'Manual Review'
  | 'Not Supported';

export interface ActionModel {
  id: string;
  step: number;
  task: string;
  parentActionId?: string | null;
  command: string;
  operation?: string | null;
  rawAction: Record<string, any>;
  attributes: Record<string, any>;
  variablesUsed: string[];
  variablesCreated: string[];
  inputs: string[];
  outputs: string[];
  children: string[];
  sourceFile: string;
  sourcePath: string;
  cloudOrDesktop: PlatformType;
  powerAutomateAction: string;
  migrationStrategy: MigrationStrategyType;
  migrationComplexity: ComplexityType;
  confidence: number;
  reason: string;
  manualSteps: string[];
  dependencies: string[];
}

export interface DisabledActionModel {
  task: string;
  file: string;
  originalStep: number;
  command: string;
  action: string;
  attributes: Record<string, any>;
  location?: string | null;
  parent?: string | null;
  reason: string;
}

export interface ActionAnalysisResponse {
  total_actions: number;
  actions: ActionModel[];
  summary_by_platform: Record<string, number>;
  summary_by_complexity: Record<string, number>;
}

File: backend/models/action.py
from typing import List, Dict, Any, Optional
from pydantic import BaseModel, Field

class DisabledActionModel(BaseModel):
    task: str
    file: str
    originalStep: int
    command: str
    action: Optional[str] = "Execute"
    attributes: Dict[str, Any] = Field(default_factory=dict)
    location: Optional[str] = None
    parent: Optional[str] = None
    reason: str = "Action was disabled in A360"

class ActionModel(BaseModel):
    id: str
    step: int
    task: str
    parentActionId: Optional[str] = None
    command: str
    operation: Optional[str] = None
    rawAction: Dict[str, Any] = Field(default_factory=dict)
    attributes: Dict[str, Any] = Field(default_factory=dict)
    variablesUsed: List[str] = Field(default_factory=list)
    variablesCreated: List[str] = Field(default_factory=list)
    inputs: List[str] = Field(default_factory=list)
    outputs: List[str] = Field(default_factory=list)
    children: List[str] = Field(default_factory=list)
    sourceFile: str = ""
    sourcePath: str = ""
    cloudOrDesktop: str = "Manual Review"  # Cloud, Desktop, Hybrid, Manual Review
    powerAutomateAction: str = "Manual Review Required"
    migrationStrategy: str = "Manual Review"  # Direct Mapping, Equivalent Connector, Cloud Replacement, Desktop Replacement, Hybrid Implementation, Custom API, Custom Script, Manual Review, Not Supported
    migrationComplexity: str = "Medium"  # Low, Medium, High
    confidence: float = 0.50
    reason: str = ""
    manualSteps: List[str] = Field(default_factory=list)
    dependencies: List[str] = Field(default_factory=list)

class ActionAnalysisResponse(BaseModel):
    total_actions: int
    actions: List[ActionModel]
    summary_by_platform: Dict[str, int]
    summary_by_complexity: Dict[str, int]
